crates/bevy_render/src/world_sync.rs

```diff
@@ -0,0 +1,261 @@
+use bevy_app::Plugin;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    observer::Trigger,
+    query::With,
+    reflect::ReflectComponent,
+    system::{Local, Query, ResMut, Resource, SystemState},
+    world::{Mut, OnAdd, OnRemove, World},
+};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_reflect::Reflect;
+use bevy_utils::tracing::warn;
+
+/// A Plugin that synchronizes entities with specific Components between the main world and the render world.
```

Comment:
```suggestion
/// A Plugin that synchronizes entities with [`SyncRenderWorld`] between the main world and the render world.
```

It's no longer true that syncing is per-component, right?